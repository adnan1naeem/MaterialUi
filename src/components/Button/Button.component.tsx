/**
 * Created by M.GhazanfarAli on 12/27/17.
 */

import * as React from 'react';
import IButton from '@blueeast/bluerain-ui-interfaces/Components/Button';
import { Button } from 'material-ui';


/**
 * The props of Button Component
 * @param {Node} props.children Label to be passed as child.
 * @param {Object} props.style style object for button's style.
 * @param {enum} props.color color prop of type enum. Default color is 'primary'.
 * @param {function} props.onClick callback function fired when button is pressed.
 * @param {boolean} props.disabled if true, renders a disabled button.
 * @param {string} props.href href prop to move to a link.
 * @param {boolean} props.fullWidth if true, button is generated with 100% width of the cnotainer.
 * @param {Node} props.iconLeft icon prop to show icon on the left side of label.
 * @param {Node} props.iconRight icon prop to show icon on the right side of label.
 * @param {boolean} props.fab if true, fab button is generated.
 * @param {boolean} props.small if true, small button is generated (small also works when fab is true).
 * @param {string} props.className string name of styling css class.
 */

 export interface ButtonMUI {
        color?: 'default' | 'accent' | 'primary' | 'contrast' | 'inherit',
        dense?: boolean,
        classes?: object,
        disableRipple?: boolean,
        disableFocusRipple?: boolean,
        component?: string | undefined,
        raised?: boolean,
 }

const BlueRainButton = (props: IButton & ButtonMUI) => {
    if(props.iconLeft){
    return (
        <Button
        raised={true || props.raised}
        color={ props.color ? (props.color ==='default'|| props.color ==='accent'|| props.color ==='primary'|| props.color ==='contrast'? props.color: 'default') : 'default'} 
        style={props.fullWidth ? (props.style, {width:'100%'}): props.style} 
        disabled={props.disabled}
        href={props.href}
        fab={props.fab}
        mini={props.small}
        dense={props.small}
        onClick={props.onClick}
        {...props}
        >
        {props.iconLeft} {props.children}
        </Button>
    );
    }
    else if(props.iconRight){
        return (
            <Button
            raised={true || props.raised}
            color={ props.color ? (props.color==='default'|| props.color==='accent'|| props.color==='primary'|| props.color==='contrast'? props.color: 'default') : 'default'} 
            style={props.fullWidth? (props.style, {width:'100%'}): props.style} 
            disabled={props.disabled}
            href={props.href}
            fab={props.fab}
            mini={props.small}
            dense={props.small}
            onClick={props.onClick}
            {...props}
            >
            {props.children} {props.iconRight}
            </Button>
        );
        }
    return (
        <Button
        raised={true || props.raised}
        color={ props.color? (props.color==='default'|| props.color==='accent'|| props.color==='primary'|| props.color==='contrast'? props.color: 'default') : 'default'} 
        style={props.fullWidth ? (props.style, {width:'100%'}): props.style} 
        disabled={props.disabled}
        href={props.href}
        fab={props.fab}
        mini={props.small}
        dense={props.small}
        onClick={props.onClick}
        {...props}
        >
        {props.children}
        </Button>
    );

}

export default BlueRainButton;