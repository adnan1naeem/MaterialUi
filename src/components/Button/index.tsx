// export { BlueRainButton as Button } from  './Button.component';