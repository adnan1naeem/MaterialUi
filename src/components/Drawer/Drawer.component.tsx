// import * as React from 'react';
// import { BlueRain, BlueRainConsumer } from '@blueeast/bluerain-os';
// import { DrawerProperties } from '@blueeast/bluerain-ui-interfaces';
// import Drawer from 'material-ui/Drawer';

// /**
//  * An interface that extends the DrawerProperties interface
//  *
//  * @export MUIDrawerProperties
//  * @interface MUIDrawerProperties
//  * @extends {DrawerProperties}
//  */
// export interface MUIDrawerProperties extends DrawerProperties {
// 	PaperProps?: any,
// 	SlideProps?: any,
// 	ModalProps?: any,
// 	variant?: 'permanent' | 'persistent' | 'temporary',
// 	classes?: any
// }

// // export const BlueRainDrawer: React.StatelessComponent<MUIDrawerProperties> = (props) => (
// // 	<MUIDrawer children={props.content} {...props}/>
// // );

// export const MUIDrawer: React.StatelessComponent<MUIDrawerProperties & { children?: React.ReactNode }> = (props) => (
// 	<BlueRainConsumer>
// 		{
// 			(BR: BlueRain) => (
// 				<BR.Components.View>
// 					<Drawer
// 						anchor={props.side}
// 						transitionDuration={props.tweenDuration}
// 						type={props.variant}
// 						{...props}
// 					>
// 						{props.children}
// 					</Drawer>
// 				</BR.Components.View>
// 			)
// 		}
// 	</BlueRainConsumer>
// );
