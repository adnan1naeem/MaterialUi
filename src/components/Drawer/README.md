# Drawer

Navigation drawers provide access to destinations in your app. Side sheets are surfaces containing supplementary content that are anchored to the left or right edge of the screen.

## Compatibility

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :----: | :---------: | :-------------: |
| ✔️     | ✖           | ✖             |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :---: | :--------: | :------------: |
| ![web image](./screenshots/Drawer.png)  |    TBD   |  TBD |

## Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| children | ReactNode | | Children of Drawer component. |
| style | object| | style object Drawer component. |
| width | number | | width of Drawer component. |
| onStateChanged | (event: object) => void | |  called on state change of Drawer component. |
| open | boolean | | open prop for Drawer component. |
| type | string | 'temporary' , 'persistent' , 'permanent'| type for Drawer component. |

## Specific Library Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| Variant | string |  'permanent' , 'persistent' , 'temporary' | Varient for Drawer Component |
| side | string |  | side prop for Drawer component |
| tweenDuration| number |  | Tween Duraion for Drawer of component. |

## How to use

```JavaScript
import React from 'react';
import BR from '@blueeast/bluerain-os';

const DrawerSample = (props) => {

    const Drawer = BR.Components.get('Drawer');
    return (
      <Drawer
        variant="persistent"
        open={true}
        tweenDuration={1000}
        side="left"
        style={AppBarStoriesStyle}
      />
    );
    }

export default DrawerSample;