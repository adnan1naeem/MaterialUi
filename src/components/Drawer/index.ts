export { MUIDrawer as Drawer, MUIDrawerProperties } from './Drawer.component';
