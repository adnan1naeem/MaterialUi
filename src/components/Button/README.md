# Button

Buttons allow users to take actions, and make choices, with a single tap.

## Compatibility

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :----: | :---------: | :-------------: |
| ✔️      | ✖            | ✖              |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :---: | :--------: | :------------: |
| ![web image](./screenshots/Button.png) |    TBD   | TBD |

## Universal Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| style | Object | | style object for button. |
| children | Node | | Content of the button component. |
| color | enum:'default' ,'inherit' ,'primary','secondary' | 'default' | Color of button component.|
| onClick | func | () => {} | Callback function fired when button is clicked. |
| disabled | bool |  | If true, renders a disabled button. |
| href | string | | href prop to move to a link. |
| fullWidth | bool |  | If true, button is generated with 100% width of the cnotainer. |
| iconLeft | Node |  | icon prop to show icon on the left side of label. |
| iconRight | Node | | icon prop to show icon on the right side of label. |
| small | bool |  | If true, renders a small sized button. |
| large | bool |  | If true, renders a large sized button.|
| variant | enum |'text' , 'flat' , 'outlined' , 'contained' , 'raised' , 'fab' , 'extendedFab'  | If true, renders a large sized button.|

## Specific Library Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| tag | func, string | button | tag prop to generate component with any other component and not with default 'button'. |
| size | enum:'small','medium','large' |  | Useful to set size of button component. |

## How to use

```JavaScript
import React from 'react';
import Button from  'bluerain-plugin-bootstrap-ui/Button';

const buttonSample = () => (
       <Button fullWidth color='primary'>Click me!</Button>
);

export default buttonSample;
```