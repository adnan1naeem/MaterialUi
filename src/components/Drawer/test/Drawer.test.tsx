import 'jest';
import   Adapter from 'enzyme-adapter-react-16';
import  React from 'react';
import  renderer from 'react-test-renderer';
import { configure, mount, shallow, } from 'enzyme';
import { Drawer } from '../../Drawer';
import { Button } from '../../Button';
configure({ adapter: new Adapter() });

test('It should return true when drawer is close. ', () => {
    const wrapper = shallow(<Drawer anchor="right" open={false} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
});
test('It should render overlay type drawer . ', () => {
    const wrapper = shallow(<Drawer anchor="right" open={false} />);
    expect(wrapper).toBeDefined();
});