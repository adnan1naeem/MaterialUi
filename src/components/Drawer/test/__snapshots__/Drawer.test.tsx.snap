// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render anchor type left .  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MUIDrawer)
    anchor="left"
    open={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anchor": "left",
      "classes": Object {
        "paparBorder": "MUIDrawer-paparBorder-1",
      },
      "open": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anchor": "left",
        "classes": Object {
          "paparBorder": "MUIDrawer-paparBorder-1",
        },
        "open": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`It should render anchor type right .  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MUIDrawer)
    anchor="right"
    open={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anchor": "right",
      "classes": Object {
        "paparBorder": "MUIDrawer-paparBorder-1",
      },
      "open": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anchor": "right",
        "classes": Object {
          "paparBorder": "MUIDrawer-paparBorder-1",
        },
        "open": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`It should render anchor type top .  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MUIDrawer)
    anchor="top"
    open={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anchor": "top",
      "classes": Object {
        "paparBorder": "MUIDrawer-paparBorder-1",
      },
      "open": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anchor": "top",
        "classes": Object {
          "paparBorder": "MUIDrawer-paparBorder-1",
        },
        "open": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`It should return bottom when drawer is bottom.  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MUIDrawer)
    anchor="bottom"
    open={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anchor": "bottom",
      "classes": Object {
        "paparBorder": "MUIDrawer-paparBorder-1",
      },
      "open": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anchor": "bottom",
        "classes": Object {
          "paparBorder": "MUIDrawer-paparBorder-1",
        },
        "open": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`It should return false when drawer is open.  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MUIDrawer)
    anchor="bottom"
    open={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anchor": "bottom",
      "classes": Object {
        "paparBorder": "MUIDrawer-paparBorder-1",
      },
      "open": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anchor": "bottom",
        "classes": Object {
          "paparBorder": "MUIDrawer-paparBorder-1",
        },
        "open": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`It should return true when drawer is bottom.  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MUIDrawer)
    anchor="bottom"
    open={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anchor": "bottom",
      "classes": Object {
        "paparBorder": "MUIDrawer-paparBorder-1",
      },
      "open": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anchor": "bottom",
        "classes": Object {
          "paparBorder": "MUIDrawer-paparBorder-1",
        },
        "open": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`It should return true when drawer is open.  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MUIDrawer)
    anchor="bottom"
    open={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anchor": "bottom",
      "classes": Object {
        "paparBorder": "MUIDrawer-paparBorder-1",
      },
      "open": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anchor": "bottom",
        "classes": Object {
          "paparBorder": "MUIDrawer-paparBorder-1",
        },
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
